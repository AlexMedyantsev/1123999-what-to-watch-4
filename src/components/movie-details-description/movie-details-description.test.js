import React from "react";
import renderer from "react-test-renderer";

import MovieDetailsDescription from './movie-details-description.jsx';


const movie = {
  image: `a`,
  posterSrc: `a`,
  bgSrc: `a`,
  title: `Name`,
  genre: `Genre`,
  year: 2014,
  score: 8.9,
  level: `level`,
  movieLink: `link`,
  previewLink: `preview-link`,
  scoresCount: 220,
  description: `description`,
  director: `director`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
  runTime: `01h30m`,
  link: `movie-page.html`,
  id: 1,
  backgroundColor: `background_color`,
  isFavorite: `is_favorite`
};

const comments = [
  {
    id: 1,
    user: {
      id: 13,
      name: `Zak`,
    },
    rating: 9,
    comment: `Unfortunately we don't have a reliable way to tell the true ratings of a movie.`,
    date: `2020-07-19T16:06:01.831Z`,
  }
];

describe(`MovieCardDescription render`, () => {
  it(`MovieCardDescription render`, () => {
    const handlerSetActive = jest.fn();
    const tree = renderer.create(
        <MovieDetailsDescription
          movie={movie}
          activeItem={`Overview`}
          setActiveItem={handlerSetActive}
          comments={comments}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
