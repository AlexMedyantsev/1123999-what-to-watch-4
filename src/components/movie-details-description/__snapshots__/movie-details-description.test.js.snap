// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardDescription render MovieCardDescription render 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item false"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item false"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      7.2
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        good
      </span>
      <span
        className="movie-rating__count"
      >
        248
         ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      The second installment of the "Fantastic Beasts" series featuring the adventures of Magizoologist Newt Scamander.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        David Yates
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Eddie Redmayne, Katherine Waterson, Dan Folger
         and other
      </strong>
    </p>
  </div>
</div>
`;
